### PR TITLE
[WIP] Add support for installing dependencies in build directory

### PR DESCRIPTION
**Pull request by kgpai**
This change allows us to download/build/install dependencies in the velox build directory. The download/install are done at cmake build time , so users do not have to preinstall dependencies. If users have dependencies preinstalled and are compatible with required dependencies then these can be used. Currently only GTest is added as an example, but it should be possible to move all dependencies this way. 

To summarize: 
 - We can have dependencies as part of our build directory
 - These will be downloaded / installed in build directory at make time
 - Support for cmake and non cmake projects
 - Supports both git repos, tar/zip files etc. 
 - Should allow us to use docker images without preinstalled dependencies. 

Should this PR be accepted , I will work on moving rest of dependencies.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -227,6 +229,10 @@ find_library(EVENT event)
 
 find_library(DOUBLE_CONVERSION double-conversion)
 
+setup_dependency(
+  PROJ GTest GIT_REPOSITORY https://github.com/google/googletest.git GIT_TAG
```
Comment (majetideepak) [inline CMakeLists.txt:233]: How can users specify `EXTERNAL_PATH`?
